Formula/libstfl.rb

```diff
@@ -23,45 +23,47 @@ class Libstfl < Formula
   depends_on "swig" => :build
   depends_on "ruby"
 
+  uses_from_macos "ncurses"
   uses_from_macos "perl"
 
   def install
-    ENV.prepend_path "PATH", Formula["python@3.9"].opt_libexec/"bin"
-
-    ENV.append "LIBS", "-lncurses -lruby"
-
     on_macos do
       ENV.append "LDLIBS", "-liconv"
-      ENV.append "LIBS", "-liconv"
-    end
+      ENV.append "LIBS", "-lncurses -lruby -liconv"
+
+      %w[
+        stfl.pc.in
+        perl5/Makefile.PL
+        ruby/Makefile.snippet
+      ].each do |f|
+        inreplace f, "ncursesw", "ncurses"
+      end
 
-    %w[
-      stfl.pc.in
-      perl5/Makefile.PL
-      ruby/Makefile.snippet
-    ].each do |f|
-      inreplace f, "ncursesw", "ncurses"
-    end
+      inreplace "stfl_internals.h", "ncursesw/ncurses.h", "ncurses.h"
 
-    inreplace "stfl_internals.h", "ncursesw/ncurses.h", "ncurses.h"
+      inreplace "Makefile" do |s|
+        s.gsub! "ncursesw", "ncurses"
+        s.gsub! "-Wl,-soname,$(SONAME)", "-Wl"
+        s.gsub! "libstfl.so.$(VERSION)", "libstfl.$(VERSION).dylib"
+        s.gsub! "libstfl.so", "libstfl.dylib"
+      end
+    end
 
-    inreplace "Makefile" do |s|
-      s.gsub! "ncursesw", "ncurses"
-      s.gsub! "-Wl,-soname,$(SONAME)", "-Wl"
-      s.gsub! "libstfl.so.$(VERSION)", "libstfl.$(VERSION).dylib"
-      s.gsub! "libstfl.so", "libstfl.dylib"
+    on_linux do
+      ENV.append "LIBS", "-lncursesw -lruby"
     end
 
+    ENV.prepend_path "PATH", Formula["python@3.9"].opt_libexec/"bin"
     xy = Language::Python.major_minor_version "python3"
     python_config = Formula["python@3.9"].opt_libexec/"bin/python-config"
```

Comment:
Python is actually not used since the checks are written in Python 2 (e.g. `python -c 'print 1'`).
Probably should drop dependency or see if possible to patch to use Python 3.

Reply:
Patched in Python 3 support and added test that fails on current macOS bottle:
```python
==> /usr/local/opt/python@3.9/bin/python3 -c import stfl
Last 15 lines from /Users/cho-m/Library/Logs/Homebrew/libstfl/test.03.python3:
2021-07-10 20:33:21 -0700

/usr/local/opt/python@3.9/bin/python3
-c
import stfl

Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'stfl'
Error: libstfl: failed
```